kyo-pure/shared/src/main/scala/kyo/aborts.scala

```diff
@@ -0,0 +1,80 @@
+package kyo
+
+import kyo.core.*
+import scala.reflect.ClassTag
+
+// Can't use an opaque type because
+// kyo-direct tests crash the compiler.
+type Aborts[-V] //= DoAbort
+
+object Aborts:
```

Comment:
I'm not sure if that's a good decision yet but I'm also considering exception handling and throwing as not pure so I've removed `Aborts.catching`, which I'm planning to reintroduce as an extension method in `kyo-core` as well. I'd love to hear thoughts on this aspects!